### ISSUE TITLE
Use <map-featurecaption> as the feature label by default

### ISSUE THREAD

**Issue by prushforth** [Maps4HTML/Web-Map-Custom-Element#947]
If a \<map-feature> contains a <map-featurecaption>, the feature should be visually labelled, using map text label defaults built into the user agent stylesheet, amended by the user stylesheet and potentially customized by the server-provided style rules found in \<map-link rel=stylesheet> and \<map-style>.  

The \<map-geometry> should be used to control the label placement algorithm.

If a feature is tiled into parts, it should be possible to merge the geometry parts using a standard attribute name which has a common value across the parts.  The label placement algorthim could use the resulting re-assembled geometry as required.

**Comment by AliyanH** [comment]
As discussed, possible overlapping problems may exist with this approach:

![image](https://github.com/Maps4HTML/Web-Map-Custom-Element/assets/55751566/9d8f333e-628b-40f6-95c0-dcb13b92e30c)

Similarly, when zoomed out, we will face the issue of improper label size since no media query exists yet to incorporate map scale which would be used to adjust size accordingly.
![image](https://github.com/Maps4HTML/Web-Map-Custom-Element/assets/55751566/3a3457dc-569a-48bb-a82a-eb89bce095f0)

**Comment by prushforth** [comment]
There are going to be challenges, but it should be possible to have a default rendering heuristic perhaps.  For example, a \<map-feature> has a `zoom` attribute aka native zoom, and perhaps we could limit the default rendering to that zoom value.

As far as overlapping labels goes, yes, this would require some label collision detection logic to be built into the renderer I guess. 

Good to keep discussing what might be involved for now.

**Comment by prushforth** [comment]
For consideration https://en.wikipedia.org/wiki/Automatic_label_placement
